tools/scitokens-admin-oauthcerts

```diff
@@ -0,0 +1,105 @@
+            "use": "sig"
+        }
+    ]
+}
+
+"""
+
+import sys
+import cryptography.utils
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.backends import default_backend
+import base64
+import binascii
+import json
+import argparse
+
+def string_from_long(data):
+    """
+    Create a base64 encoded string for an integer
+    """
+    return base64.urlsafe_b64encode(cryptography.utils.int_to_bytes(data)).decode('ascii')
+
+def bytes_from_long(data):
+    """
+    Create a base64 encoded bytes for an integer
+    """
+    return base64.urlsafe_b64encode(cryptography.utils.int_to_bytes(data))
+
+
+def add_args():
+    """
+    Generate the ArgumentParser object for the CLI.
+    """
+    parser = argparse.ArgumentParser(description='Create a public key for the oauth certs file')
+    
+    # Mutual exclude.  Either give the private keyfile, or public, but not both, and at least 1
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('--private-keyfile',
+                        help='Location of the unencrypted private key file')
+    group.add_argument('--public-keyfile',
+                        help='Location of a public key file')
+
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    """
+    Given a set of command line parameters, generate a corresponding oauth2 certs file.
+    """
+    args = add_args()
+    
+    if args.private_keyfile != None:
+        with open(args.private_keyfile, 'rb') as key_file:
+            private_key = serialization.load_pem_private_key(
+                key_file.read(),
+                password=None,
+                backend=default_backend()
+            )
+            public_key = private_key.public_key()
+    else:
+        with open(args.public_keyfile, "rb") as key_file:
+            public_key = serialization.load_pem_public_key(
+                key_file.read(),
+                backend=default_backend()
+            )
+
+    # Get the public numbers
+    #public_key = private_key.public_key()
+    numbers = public_key.public_numbers()
+
+    # Hash the public "n", and use it for the Key ID (kid)
+    digest = hashes.Hash(hashes.SHA256(), backend=default_backend())
+    digest.update(bytes_from_long(numbers.n))
+    kid = binascii.hexlify(digest.finalize())
+
+    keys = {'keys': [
+        {
+            "alg": "RS256",
+            "n": string_from_long(numbers.n),
+            "e": string_from_long(numbers.e),
+            "kty": "RSA",
+            "use": "sig",
+            "kid": kid.decode('utf-8')
+        }
+    ]}
+
+    print(json.dumps(keys, sort_keys=True,
```

Comment:
This tool makes me think of https://github.com/opensciencegrid/scitokens-credmon/issues/4; that ticket also requires one to print out a JWKS-style key.

Can we refactor most of the `main` logic into a utility function that returns a python dictionary?  The `main` function here would just do argument parsing to get a public key, call the utility function, then convert the returned dictionary into a JSON string?

Reply:
It's not really printing out anything with JWTs.  These are specially formatted json for OAuth issuers.  I don't see a bunch of overlap from the credmon and this tool.